# Laporan Analisis Data: Regresi Linier Sederhana

Proyek ini menerapkan analisis Regresi Linier Sederhana untuk menentukan hubungan antara variabel independen ($X$) dan variabel dependen ($Y$) berdasarkan sebaran data dari grafik GeoGebra. Model diselesaikan dan divalidasi menggunakan dua pendekatan: **Perhitungan Analitik (Metode Matriks)** dan **Implementasi Kode Python (Scikit-Learn & Numpy)**.

---

## 1. Dataset (Titik Observasi dari GeoGebra)

Data mentah diekstraksi secara langsung dari koordinat titik yang tertera pada panel aljabar grafik GeoGebra:

| Titik | Sumbu X (Fitur) | Sumbu Y (Target) |
|:-----:|:---------------:|:----------------:|
|   A   |        2        |        2         |
|   B   |        4        |        3         |
|   C   |        5        |        5         |
|   D   |        3        |        4         |
|   E   |        3        |        3         |
|   F   |        4        |        5         |
|   G   |        5        |        6         |

---

## 2. Perhitungan Analitik (Metode Matriks)

Persamaan dasar aljabar linier yang digunakan untuk mencari estimator parameter koefisien regresi adalah persamaan *Ordinary Least Squares* (OLS):

$$\hat{\beta} = (X^T X)^{-1} X^T Y$$

Dimana vektor kolom $\hat{\beta}$ menampung parameter model:

$$
\hat{\beta} = \begin{bmatrix} \beta_0 \\ \beta_1 \end{bmatrix}
$$

* $\beta_0$ = *Intercept* (Titik potong sumbu Y)
* $\beta_1$ = *Slope* (Koefisien kemiringan arah variabel $X$)

### Langkah 1: Menyusun Matriks Desain $X$ dan Vektor Target $Y$
Kolom pertama pada matriks desain $X$ diisi dengan elemen bernilai $1$ untuk mengakomodasi perhitungan nilai konstanta *intercept* ($\beta_0$).

$$X = \begin{bmatrix} 1 & 2 \\ 1 & 4 \\ 1 & 5 \\ 1 & 3 \\ 1 & 3 \\ 1 & 4 \\ 1 & 5 \end{bmatrix}, \quad Y = \begin{bmatrix} 2 \\ 3 \\ 5 \\ 4 \\ 3 \\ 5 \\ 6 \end{bmatrix}$$

### Langkah 2: Menghitung Transpose Matriks $X$ ($X^T$)
$$X^T = \begin{bmatrix} 1 & 1 & 1 & 1 & 1 & 1 & 1 \\ 2 & 4 & 5 & 3 & 3 & 4 & 5 \end{bmatrix}$$

### Langkah 3: Menghitung Perkalian Matriks $X^T X$
Hasil perkalian menghasilkan matriks simetris berukuran $2 \times 2$:

$$X^T X = \begin{bmatrix} 7 & 26 \\ 26 & 104 \end{bmatrix}$$

### Langkah 4: Menghitung Invers Matriks $(X^T X)^{-1}$
$$(X^T X)^{-1} = \frac{1}{(7 \times 104) - (26 \times 26)} \begin{bmatrix} 104 & -26 \\ -26 & 7 \end{bmatrix}$$
$$(X^T X)^{-1} = \frac{1}{52} \begin{bmatrix} 104 & -26 \\ -26 & 7 \end{bmatrix} = \begin{bmatrix} 2.0000 & -0.5000 \\ -0.5000 & 0.1346 \end{bmatrix}$$

### Langkah 5: Menghitung Perkalian Vektor Sisi Kanan $X^T Y$
$$X^T Y = \begin{bmatrix} 2+3+5+4+3+5+6 \\ (2\times2)+(4\times3)+(5\times5)+(3\times4)+(3\times3)+(4\times5)+(5\times6) \end{bmatrix} = \begin{bmatrix} 28 \\ 112 \end{bmatrix}$$

### Langkah 6: Kalkulasi Akhir Estimator Koefisien ($\hat{\beta}$)
$$\hat{\beta} = \begin{bmatrix} 2.0000 & -0.5000 \\ -0.5000 & 0.1346 \end{bmatrix} \begin{bmatrix} 28 \\ 112 \end{bmatrix} = \begin{bmatrix} 0.0000 \\ 1.0769 \end{bmatrix}$$

Melalui kalkulasi ini, didapatkan parameter model analitik:
* **$\beta_0$ (Intercept)** = `0.0000`
* **$\beta_1$ (Koefisien X)** = `1.0769`

---

## 3. Implementasi Kode Python (Pengecekan & Validasi)

Berikut adalah *script* Python untuk melakukan pengecekan silang (*cross-check*) otomatis menggunakan library `scikit-learn` dan kalkulasi matriks `numpy`:

In [1]:
import numpy as np
from sklearn.linear_model import LinearRegression

# 1. Input Data Sesuai Koordinat GeoGebra
X = np.array([2, 4, 5, 3, 3, 4, 5]).reshape(-1, 1) 
Y = np.array([2, 3, 5, 4, 3, 5, 6])

print("= = = = = VALIDASI KODE PYTHON = = = = =")

# 2. Pengecekan Menggunakan Scikit-Learn
model = LinearRegression()
model.fit(X, Y)

print("[SKLEARN] Hasil Estimasi:")
print(f"  - Intercept (b0) : {model.intercept_:.4f}")
print(f"  - Koefisien (b1) : {model.coef_[0]:.4f}\n")

# 3. Pengecekan Menggunakan Komputasi Matriks Numpy (Analitik)
X_mat = np.c_[np.ones((X.shape[0], 1)), X] 
Y_mat = Y.reshape(-1, 1)

beta_hat = np.linalg.inv(X_mat.T.dot(X_mat)).dot(X_mat.T.dot(Y_mat))

print("[NUMPY MATRIKS] Hasil Estimasi Analitik:")
print(f"  - Intercept (b0) : {beta_hat[0][0]:.4f}")
print(f"  - Koefisien (b1) : {beta_hat[1][0]:.4f}")

= = = = = VALIDASI KODE PYTHON = = = = =
[SKLEARN] Hasil Estimasi:
  - Intercept (b0) : -0.0000
  - Koefisien (b1) : 1.0769

[NUMPY MATRIKS] Hasil Estimasi Analitik:
  - Intercept (b0) : 0.0000
  - Koefisien (b1) : 1.0769


## 4. Panduan Step-by-Step Mendapatkan Hasil di GeoGebra

Untuk memvalidasi data di atas dan memunculkan garis regresi linier terbaik secara otomatis di GeoGebra, ikuti langkah-langkah berikut:

### Langkah 1: Input Titik Koordinat
Masukkan ketujuh titik koordinat ke dalam *Input Bar* di sebelah kiri GeoGebra satu per satu dengan format berikut, lalu tekan **Enter**:
* `A = (2, 2)`
* `B = (4, 3)`
* `C = (5, 5)`
* `D = (3, 4)`
* `E = (3, 3)`
* `F = (4, 5)`
* `G = (5, 6)`

### Langkah 2: Membuat List dari Titik-Titik Data
Garis regresi di GeoGebra membutuhkan sebuah *list* objek sebagai parameternya. Buat daftar titik tersebut dengan mengetik perintah berikut di *Input Bar*:
```text
L_1 = {A, B, C, D, E, F, G}
```
GeoGebra akan mengelompokkan semua titik ke dalam satu variabel bernama $L_1$.

### Langkah 3: Menggunakan Perintah Regresi(FitLine)
Untuk menggambar garis tren linear terbaik (best-fit line) berdasarkan data di $L_1$, ketik perintah regresi linier berikut di Input Bar:
```
Plaintextf = FitLine(L_1)
```
### Langkah 4: Membaca Hasil Nilai Output
Setelah menekan Enter, GeoGebra akan langsung menampilkan garis lurus pada grafik dan memunculkan persamaan fungsi pada panel aljabar di sebelah kiri:
```
Plaintextf(x) = 1.08x + 0
```

## Gambar Hasil Geogebra

![hasil](gambar/gebra.png)

(Catatan: GeoGebra secara default membulatkan angka desimal menjadi 2 angka di belakang koma, di mana $1.0769$ dibulatkan menjadi $1.08$ dan intercept $-0.0000...$ ditulis sebagai $0$).

## 5. Kesimpulan Kepaduan Data
Berdasarkan perbandingan tiga metode, hasil yang didapatkan sepenuhnya sinkron dan konsisten:
1. Analitik & Python menghasilkan presisi tinggi: $Intercept = 0.0000$ dan $Slope = 1.0769$.
2. GeoGebra memberikan visualisasi grafis fiting garis yang tepat melintasi koordinat pusat $(0,0)$ dengan pembulatan fungsi $f(x) = 1.08x$.Formulasinya terbukti sah dan valid:

$$\mathbf{Y = 1.0769 X}$$